tasks/backfill_gh_app_installations.py

```diff
@@ -0,0 +1,192 @@
+        db_session: Session,
+        ownerid: int,
+        owner_service: torngit.base.TorngitBaseAdapter,
+        gh_app_installation: GithubAppInstallation,
+    ):
+        repos = await owner_service.list_repos_using_installation()
+        if repos:
+            # Fetching all repos service ids we have for that owner in the DB
+            repo_service_ids_in_db = [
+                repo.service_id
+                for repo in db_session.query(Repository.service_id)
+                .filter_by(ownerid=ownerid)
+                .all()
+            ]
+
+            # Add service ids from provider that we have DB records for to a list
+            new_repo_service_ids = []
+            for repo in repos:
+                repo_data = repo["repo"]
+                service_id = repo_data["service_id"]
+                if service_id and service_id in repo_service_ids_in_db:
+                    new_repo_service_ids.append(service_id)
+            log.info(
+                "Added the following repo service ids to this gh app installation",
+                extra=dict(
+                    ownerid=ownerid,
+                    installation_id=gh_app_installation.installation_id,
+                    new_repo_service_ids=list(new_repo_service_ids),
+                ),
+            )
+            gh_app_installation.repository_service_ids = new_repo_service_ids
+            db_session.commit()
+
+    async def backfill_existing_gh_apps(
+        self, db_session: Session, owner_ids: List[int] = None
+    ):
+        # Get owners that have installations, and installations queries
+        owners_query = (
+            db_session.query(Owner)
+            .join(GithubAppInstallation)
+            .filter(
+                Owner.service == "github",
+                Owner.ownerid == GithubAppInstallation.ownerid,
+            )
+        )
+        gh_app_installations_query = db_session.query(GithubAppInstallation)
+
+        # Filter if owner_ids were provided
+        if owner_ids:
+            owners_query = owners_query.filter(Owner.ownerid.in_(owner_ids))
+            gh_app_installations_query = gh_app_installations_query.filter(
+                GithubAppInstallation.ownerid.in_(owner_ids)
+            )
+
+        owners: List[Owner] = owners_query.all()
+        gh_app_installations: List[
+            GithubAppInstallation
+        ] = gh_app_installations_query.all()
+
+        # I need to make reference of these in the gh installation app, Ill take suggestions here
+        owners_dict = {owner.ownerid: owner for owner in owners}
+
+        for gh_app_installation in gh_app_installations:
+            # Check if gh app has 'all' repositories selected
+            installation_id = gh_app_installation.installation_id
+
+            owner = owners_dict[gh_app_installation.ownerid]
+            ownerid = owner.ownerid
+
+            owner_service = get_owner_provider_service(
+                owner=owner, using_integration=True
```

Comment:
I had to get all the owners + ownerid cause I needed them per gh app installation to use this service